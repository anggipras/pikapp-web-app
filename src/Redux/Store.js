import { createStore } from 'redux'
// import Thunk from 'redux-thunk'
import Reducers from './Reducers'

const store = createStore(Reducers)

export default store