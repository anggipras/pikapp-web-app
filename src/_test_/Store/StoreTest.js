import Axios from "axios";
import { address, clientId } from "../../Asset/Constant/APIConstant";
import { v4 as uuidV4 } from "uuid";

export const fetchData = async () => {
    let latitude = -6.28862
    let longitude = 106.71789
    let addressRoute = address + "home/v2/merchant/" + longitude + "/" + latitude + "/ALL/"

    let uuid = uuidV4();
    uuid = uuid.replace(/-/g, "");
    const date = new Date().toISOString();

    const allMerchantAPI = await Axios(addressRoute, {
        headers: {
            "Content-Type": "application/json",
            "x-request-id": uuid,
            "x-request-timestamp": date,
            "x-client-id": clientId,
            "token": "PUBLIC",
            "category": "1",
        },
        method: "GET",
        params: {
            page: 0,
            size: 6
        }
    })

    let allData = allMerchantAPI.data.results
    let realData = []
    realData.push({
        address: allData.merchant_address,
        rating: allData.merchant_rating,
        logo: allData.merchant_logo,
        distance: allData.merchant_distance,
        storeId: allData.mid,
        storeName: allData.merchant_name,
        storeDesc: "",
        storeImage: allData.merchant_pict,
    })

    return realData
}

export const loadMoreMerch = () => {
    let idCol = 0
    let page = 0 //will load more merchant until all pages load
    let totalpage = 7 //there are 6 pages for instance

    do {
        if (idCol <= page) {
            idCol += 1
            page += 1
            console.log('loadagain');
        } else {
            console.log('disable scroll');
        }
    } while (page < totalpage - 1);

    return page
}