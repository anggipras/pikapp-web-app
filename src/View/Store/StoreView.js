import React from "react";
import StarIcon from '../../Asset/Icon/star.png'
import LocaIcon from '../../Asset/Icon/location.png'
import queryString from "query-string";
import { Link } from "react-router-dom";
import { address, clientId, googleKey } from "../../Asset/Constant/APIConstant";
import { v4 as uuidV4 } from "uuid";
import Axios from "axios";
import Cookies from "js-cookie"
import Geocode from "react-geocode"
import Skeleton from 'react-loading-skeleton'
import { connect } from 'react-redux'
import { DoneLoad, IsMerchantQR } from '../../Redux/Actions'
import TourPage from '../../Component/Tour/TourPage';

class StoreView extends React.Component {
  state = {
    page: 0,
    size: 6,
    location: "",
    data: {
      title: "",
      image: "",
      desc: "",
      data: [
        {
          address: "",
          rating: "",
          logo: "",
          distance: "",
          storeId: "",
          storeName: "",
          storeDesc: "",
          storeImage: "",
        },
      ],
    },
    idCol: 0,
    totalPage: 0,
    boolpage: false,
    loadView: true,
    allMerchantAPI: [],
    lat: "",
    lon: "",
    startTour : false,
    steptour:[
      {
        selector: '',
        content : () => (
          <div>
            <h2>Selamat Datang di PikApp!</h2>
            <br />
            <span>Yuk caritau cara memesan dengan PikApp dengan mudah.</span>
          </div>
        )
      },
      {
        selector: '.merchantList-layout',
        content : () => (
          <div>
            <h4>Ini adalah Daftar Restoran</h4>
            <br />
            <span>Kamu bisa cek restoran apa saja yang ada di dalam foodcourt ini. Anda bisa tap salah satu restoran untuk ke halaman berikutnya.</span>
          </div>
        )
      }
    ]
  };

  componentDidMount() {
    this.props.DoneLoad()
    this.props.IsMerchantQR(false);
    Cookies.set("homePage", window.location.search)
    var auth = {
      isLogged: false,
      token: "",
      new_event: true,
      recommendation_status: false,
      email: "",
    };
    if (Cookies.get("auth") !== undefined) {
      auth = JSON.parse(Cookies.get("auth"))
    } 

    if (localStorage.getItem("storeTour") == 1) {
      this.setState({ startTour : true});
    }

    // else {
    const value = queryString.parse(window.location.search);
    var merchant = "";

    // if (navigator.geolocation) { //SHUTDOWN FOR A WHILE
    //   navigator.geolocation.getCurrentPosition(position => {
    //     let latitude = 0;
    //     let longitude = 0;
    
    //     if(position) {
    //       let latitude = position.coords.latitude
    //       let longitude = position.coords.longitude
    //       let longlat = { lat: latitude, lon: longitude }
    //       console.log(latitude, longitude);
    //       this.setState({ lat: latitude, lon: longitude })
    //       localStorage.setItem("longlat", JSON.stringify(longlat))
    //     } else {
    //       let latitude = 1;
    //       let longitude = 1;
    //       let longlat = { lat: latitude, lon: longitude }
    //       console.log(latitude, longitude);
    //       this.setState({ lat: latitude, lon: longitude })
    //       localStorage.setItem("longlat", JSON.stringify(longlat))
    //     }
    //   });
    // }

    let latitude = -6.28862
    let longitude = 106.71789
    let longlat = { lat: latitude, lon: longitude }
    console.log(latitude, longitude);
    this.setState({ lat: latitude, lon: longitude })
    localStorage.setItem("longlat", JSON.stringify(longlat))
    
    // Show a map centered at latitude / longitude.

    if (localStorage.getItem("longlat")) {
      var getLocation = JSON.parse(localStorage.getItem("longlat"))
      latitude = getLocation.lat
      longitude = getLocation.lon
    } else {
      // window.location.href = "/login"
    }

    if (auth.isLogged === false) {
      var lastLink = { value: window.location.href }
      Cookies.set("lastLink", lastLink, { expires: 1 })
      // window.location.href = "/login"
    }
    else {
      longitude = value.longitude || longitude
      latitude = value.latitude || latitude
      if (window.location.href.includes('?latitude') || window.location.href.includes('store?')) {

      } else {
        window.location.href = window.location.href + `?latitude=${latitude}&longitude=${longitude}`
      }
    }
    longitude = value.longitude || longitude
    latitude = value.latitude || latitude
    merchant = value.merchant;

    // GOOGLE GEOCODE
    if (localStorage.getItem("address")) {
      var getAdress = JSON.parse(localStorage.getItem("address"))
      this.setState({ location: getAdress })
    } else {
      Geocode.setApiKey(googleKey)
      Geocode.fromLatLng(latitude, longitude)
        .then((res) => {
          console.log(res.results[0].formatted_address);
          this.setState({ location: res.results[0].formatted_address })
          localStorage.setItem("address", JSON.stringify(res.results[0].formatted_address));
        })
        .catch((err) => {
          this.setState({ location: "Tidak tersedia" })
        })
    }

    let addressRoute;
    if (merchant === undefined) {
      addressRoute =
        address + "home/v2/merchant/" + longitude + "/" + latitude + "/ALL/";
    } else {
      addressRoute =
        address +
        "home/v1/merchant/" +
        longitude +
        "/" +
        latitude +
        "/" +
        merchant
        + "/"
    }
    var stateData;
    let uuid = uuidV4();
    uuid = uuid.replace(/-/g, "");
    const date = new Date().toISOString();
    Axios(addressRoute, {
      headers: {
        "Content-Type": "application/json",
        "x-request-id": uuid,
        "x-request-timestamp": date,
        "x-client-id": clientId,
        "token": "PUBLIC",
        "category": "1",
      },
      method: "GET",
      params: {
        page: this.state.page,
        size: this.state.size
      }
    })
      .then((res) => {
        console.log(res.data.results);
        stateData = { ...this.state.data };
        let responseDatas = res.data;
        stateData.data.pop();
        responseDatas.results.forEach((data) => {
          stateData.data.push({
            address: data.merchant_address,
            rating: data.merchant_rating,
            logo: data.merchant_logo,
            distance: data.merchant_distance,
            storeId: data.mid,
            storeName: data.merchant_name,
            storeDesc: "",
            storeImage: data.merchant_pict,
          })
        })
        if (Cookies.get("fcaddress") !== undefined) {
          let foodcourtadd = Cookies.get("fcaddress")
          let filterMerchantDetail = stateData.data.filter(fcVal => {
            return fcVal.address.toLocaleLowerCase().includes(foodcourtadd.toLocaleLowerCase())
          })
          stateData.data = filterMerchantDetail
          let filterMerchantMain = res.data.results.filter(fcVal => {
            return fcVal.merchant_address.toLocaleLowerCase().includes(foodcourtadd.toLocaleLowerCase())
          })
          this.setState({ data: stateData, loadView: false, totalPage: responseDatas.total_pages, allMerchantAPI: filterMerchantMain });
        } else {
          this.setState({ data: stateData, loadView: false, totalPage: responseDatas.total_pages, allMerchantAPI: res.data.results });
        }

        document.addEventListener('scroll', this.loadMoreMerchant)
        if (Cookies.get("fcaddress") === undefined) {
          if (res.data.results.length < 6) {
            document.removeEventListener('scroll', this.loadMoreMerchant)
          }
        }
      })
      .catch((err) => {
      });
  }

  componentDidUpdate() {
    if (this.state.idCol > 0) {
      if (this.state.boolpage === true) {
        let addressRoute = address + "home/v2/merchant/" + this.state.lon + "/" + this.state.lat + "/ALL/";
        var stateData;
        let uuid = uuidV4();
        uuid = uuid.replace(/-/g, "");
        const date = new Date().toISOString();
        Axios(addressRoute, {
          headers: {
            "Content-Type": "application/json",
            "x-request-id": uuid,
            "x-request-timestamp": date,
            "x-client-id": clientId,
            "token": "PUBLIC",
            "category": "1",
          },
          method: "GET",
          params: {
            page: this.state.page,
            size: this.state.size
          }
        })
          .then((res) => {
            stateData = { ...this.state.data };
            let responseDatas = res.data;
            if (Cookies.get("fcaddress") !== undefined) {
              let foodcourtadd = Cookies.get("fcaddress")
              let filterMerchantMain = res.data.results.filter(fcVal => {
                return fcVal.merchant_address.toLocaleLowerCase().includes(foodcourtadd.toLocaleLowerCase())
              })

              filterMerchantMain.forEach((data) => {
                stateData.data.push({
                  address: data.merchant_address,
                  rating: data.merchant_rating,
                  logo: data.merchant_logo,
                  distance: data.merchant_distance,
                  storeId: data.mid,
                  storeName: data.merchant_name,
                  storeDesc: "",
                  storeImage: data.merchant_pict,
                })
              })

              let updateMerchant = [...this.state.allMerchantAPI]
              filterMerchantMain.forEach((data) => {
                updateMerchant.push(data)
              })
              this.setState({ boolpage: false, allMerchantAPI: updateMerchant })
            } else {
              responseDatas.results.forEach((data) => {
                stateData.data.push({
                  address: data.merchant_address,
                  rating: data.merchant_rating,
                  logo: data.merchant_logo,
                  distance: data.merchant_distance,
                  storeId: data.mid,
                  storeName: data.merchant_name,
                  storeDesc: "",
                  storeImage: data.merchant_pict,
                })
              })
              let updateMerchant = [...this.state.allMerchantAPI]
              responseDatas.results.forEach((data) => {
                updateMerchant.push(data)
              })
              this.setState({ boolpage: false, allMerchantAPI: updateMerchant })
            }

            document.addEventListener('scroll', this.loadMoreMerchant)
            if (this.state.page === this.state.totalPage - 1) {
              this.setState({ idCol: this.state.idCol + 1 })
            }
          })
          .catch((err) => {
          });
      }
    }
  }

  storeClick = (e) => {
    localStorage.setItem('page', JSON.stringify(1))
  }
  handleDetail(data) {
    return <Link to={"/status"}></Link>;
  }

  isBottom = (el) => {
    return (el.getBoundingClientRect().top + 50) <= window.innerHeight
  }

  loadMoreMerchant = () => {
    const wrappedElement = document.getElementById("idCol")
    if (this.state.idCol <= this.state.page) {
      if (this.isBottom(wrappedElement)) {
        // console.log('testloadmore');
        this.setState({ idCol: this.state.idCol + 1, page: this.state.page + 1, boolpage: true })
        document.removeEventListener('scroll', this.loadMoreMerchant)
      }
    } else {
      document.removeEventListener('scroll', this.loadMoreMerchant)
    }
  }

  componentWillUnmount() {
    document.removeEventListener('scroll', this.loadMoreMerchant)
  }

  merchantLoading = () => (
    <div className='merchantList-layout' >
      <Skeleton style={{ paddingTop: 100, width: "100%", height: "100%" }} />
    </div>
  )

  tourPage = () => {
    if (this.state.startTour === true) {
      return (
        <TourPage 
          stepsContent={this.state.steptour}
          isShowTour={this.state.startTour}
          isHideTour={() =>this.showTourPage(false)}
        />
      )
    }
  }

  showTourPage = (isShowTour) => {
    this.setState({ startTour: isShowTour });
    document.body.style.overflowY = 'auto';
    localStorage.setItem('storeTour', 0);
  }

  render() {
    if (localStorage.getItem('page')) {
      let currentPage = JSON.parse(localStorage.getItem('page'))
      if (currentPage === 1) {
        localStorage.setItem('page', JSON.stringify(0))
        window.location.reload()
      }
    }
    // console.log(this.state.data.data);
    const storeDatas = this.state.data.data.map((data) => {
      return data;
    });
    var allCards = storeDatas.map((cardData, indexCard) => {
      return (
        <Link to={"/store?mid=" + cardData.storeId} style={{ textDecoration: "none" }} onClick={() => this.storeClick(cardData)} >
          <div key={indexCard} className='merchantList-layout' data-testid="merchantlist-item">
            <div className='merchantList-banner'>
              {
                cardData.storeImage ?
                  <img src={cardData.storeImage} className='merchantList-image' alt='' />
                  :
                  <Skeleton style={{ paddingTop: 100, width: "100%", height: "100%" }} />
              }
            </div>
            
            <div className='merchantList-content'>
              <div className='merchantList-contentLocStar'>
                <div className='merchantList-ratingArea'>
                  <img src={StarIcon} className='merchantList-ratingIcon' alt='' />
                  <div className='merchantList-ratingScore'>{cardData.rating ? cardData.rating : "5.0"}</div>
                </div>

                <div className='merchantList-locArea'>
                  <img src={LocaIcon} className='merchantList-locIcon' alt='' />
                  {
                    cardData.distance ?
                      <div className='merchantList-location'>{cardData.distance} <span className='merchantList-distance'>{'(' + Math.round((parseInt(cardData.distance) / 22) * 60) + ' min)'}</span></div>
                      :
                      <Skeleton style={{ paddingTop: 10, width: "100%", height: "100%" }} />
                  }
                </div>
              </div>

              {
                cardData.storeName ?
                  <div className='merchantList-storeName'>{cardData.storeName}</div>
                  :
                  <Skeleton style={{ paddingTop: 10, width: "100%", height: "100%" }} />
              }

              {/* <div className='merchantList-storeCategory'>Merchant Categories</div> */}
            </div>
          </div>
        </Link>
      );
    });

    return (
      <div className='merchantList-background'>
        {/* <div className="storeColumn">
          <h6 className="" style={{ textAlign: "left" }}>
            Lokasi:
            </h6>
          <p className="storeLabel" style={{ textAlign: "left" }}>
            {this.state.location || <Skeleton height={20} />}
          </p>
        </div> */}
        <div>
          <div className='merchantList-grid'>
            {allCards}
          </div>
          {
            !this.state.loadView ?
              this.state.idCol <= this.state.page ?
                this.state.totalPage - 1 === this.state.page ?
                  null
                  :
                  <div id={"idCol"}>
                    {this.merchantLoading()}
                  </div>
                :
                null
              :
              null
          }
          {this.tourPage()}
        </div>
      </div>
    );
  }
}

export default connect(null, { DoneLoad, IsMerchantQR })(StoreView)
