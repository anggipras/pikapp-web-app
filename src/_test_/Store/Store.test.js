import { shallow } from "enzyme";
import { Provider } from 'react-redux'
import Store from '../../Redux/Store'
import StoreView from "../../View/Store/StoreView";
import Axios from "axios";
import { fetchData } from './StoreTest'
import { BrowserRouter as Router } from "react-router-dom";
import Store from '../../Redux/Store'
import StoreView from "../../View/Store/StoreView";
import { fetchData, loadMoreMerch } from './StoreTest'
import renderer from 'react-test-renderer'
import Store from '../../Redux/Store'
import StoreView from "../../View/Store/StoreView";
import Axios from "axios";
import { fetchData } from './StoreTest'

it("renders without crashing StoreView", () => {
  shallow(
    <Provider store={Store}>
      <StoreView />
    </Provider>
  );
});

jest.mock('Axios')

describe('fetchData', () => {
  it('render merchant list data correctly', () => {
    let testData = [
      {
        "address": "Jl. Regensi Melati Mas F10/2",
        "logo": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/ropang-logo.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=31c6c031d30e92070634b5637eb3ee0d26af4009fc517a6c3af6de907a041d1a",
        "distance": "7.61 km",
        "storeId": "M00000008",
        "storeName": "Pikapp Store",
        "storeDesc": "",
        "storeImage": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/ropang-logo.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43199&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=2c65147879f351eca758aeadec4d465750383d975640815f3973894717edbe7f"
      },
      {
        "address": "Jl. Regensi Melati Mas F10/2",
        "logo": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/ropang-logo.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=31c6c031d30e92070634b5637eb3ee0d26af4009fc517a6c3af6de907a041d1a",
        "distance": "7.61 km",
        "storeId": "M00000009",
        "storeName": "Andrew Meat Shop",
        "storeDesc": "",
        "storeImage": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/ropang-logo.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43199&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=2c65147879f351eca758aeadec4d465750383d975640815f3973894717edbe7f"
      },
      {
        "address": "Jl. Regensi Melati Mas F10/2",
        "logo": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/photo-1552566626-52f8b828add9.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=8d45ac4fde90997792f18e7b200543172a0ada25e754b17ff006ca22f71c57e4",
        "distance": "7.61 km",
        "storeId": "M00000010",
        "storeName": "Dummy Store 1",
        "storeDesc": "",
        "storeImage": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/photo-1552566626-52f8b828add9.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=8d45ac4fde90997792f18e7b200543172a0ada25e754b17ff006ca22f71c57e4"
      },
      {
        "address": "Jl. Regensi Melati Mas F10/2",
        "logo": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/photo-1514933651103-005eec06c04b.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=e23b68cdc2dec8d3025b4c9b5ada073b60e5e2fec5c18234744388ec151d372b",
        "distance": "7.61 km",
        "storeId": "M00000011",
        "storeName": "Dummy Store 2",
        "storeDesc": "",
        "storeImage": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/photo-1514933651103-005eec06c04b.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=e23b68cdc2dec8d3025b4c9b5ada073b60e5e2fec5c18234744388ec151d372b"
      },
      {
        "address": "Jl. Regensi Melati Mas F10/2",
        "logo": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/photo-1414235077428-338989a2e8c0.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=39b67d9fdd738dd0a8399225000bd291943ad4b637ee07e627c280af8793c2af",
        "distance": "7.61 km",
        "storeId": "M00000012",
        "storeName": "Dummy Store 3",
        "storeDesc": "",
        "storeImage": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/photo-1414235077428-338989a2e8c0.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=39b67d9fdd738dd0a8399225000bd291943ad4b637ee07e627c280af8793c2af"
      },
      {
        "address": "Jl. Regensi Melati Mas F10/2",
        "logo": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/photo-1552566626-52f8b828add9.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=8d45ac4fde90997792f18e7b200543172a0ada25e754b17ff006ca22f71c57e4",
        "distance": "7.61 km",
        "storeId": "M00000013",
        "storeName": "Dummy Store 4",
        "storeDesc": "",
        "storeImage": "https://production-merchant.s3.ap-southeast-1.amazonaws.com/photo-1552566626-52f8b828add9.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210518T080323Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=AKIAYBXTVRKM5MEIYZEG%2F20210518%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Signature=8d45ac4fde90997792f18e7b200543172a0ada25e754b17ff006ca22f71c57e4"
      }
    ]

    fetchData().then(res => {
      expect(res).toEqual(testData)
    })

  })
})
      let allData = res.data.results
      let realData = []
      realData.push({
        address: allData.merchant_address,
        rating: allData.merchant_rating,
        logo: allData.merchant_logo,
        distance: allData.merchant_distance,
        storeId: allData.mid,
        storeName: allData.merchant_name,
        storeDesc: "",
        storeImage: allData.merchant_pict,
      })

      expect(realData).toEqual(testData)
  })

  it('matches the snapshot', () => {
    const tree = renderer.create(
      <Router>
        <Provider store={Store}>
          <StoreView />
        </Provider>
      </Router>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

it('test loadmoremerchant', () => {
  expect(loadMoreMerch()).toBe(6)
<<<<<<< HEAD
<<<<<<< HEAD
})
>>>>>>> b88195ce4784105b65b179d6115d42e51ea5fe2a
=======
})
>>>>>>> 68f3f0b4fa8679d082e885ebbc981032b4320121
=======
})
=======
      expect(res).toEqual(testData)
    })

  })
})
>>>>>>> mocking async fetch merchant list page
>>>>>>> 70a947b847c5fca97ac1ab64eb9fa5e5018e10ac
