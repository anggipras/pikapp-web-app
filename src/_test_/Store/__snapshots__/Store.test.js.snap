// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot StoreView 1`] = `
<div
  className="merchantList-background"
>
  <div>
    <div
      className="merchantList-grid"
    >
      <a
        href="/store?mid="
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="merchantList-layout"
          data-testid="merchantlist-item"
        >
          <div
            className="merchantList-banner"
          >
            <span>
              <span
                className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                style={
                  Object {
                    "height": "100%",
                    "paddingTop": 100,
                    "width": "100%",
                  }
                }
              >
                ‌
              </span>
            </span>
          </div>
          <div
            className="merchantList-content"
          >
            <div
              className="merchantList-contentLocStar"
            >
              <div
                className="merchantList-ratingArea"
              >
                <img
                  alt=""
                  className="merchantList-ratingIcon"
                  src="star.png"
                />
                <div
                  className="merchantList-ratingScore"
                >
                  5.0
                </div>
              </div>
              <div
                className="merchantList-locArea"
              >
                <img
                  alt=""
                  className="merchantList-locIcon"
                  src="location.png"
                />
                <span>
                  <span
                    className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                    style={
                      Object {
                        "height": "100%",
                        "paddingTop": 10,
                        "width": "100%",
                      }
                    }
                  >
                    ‌
                  </span>
                </span>
              </div>
            </div>
            <span>
              <span
                className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                style={
                  Object {
                    "height": "100%",
                    "paddingTop": 10,
                    "width": "100%",
                  }
                }
              >
                ‌
              </span>
            </span>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
